### DISCUSSION TITLE
Autosave for unsaved files

### DISCUSSION THREAD

**Discussion by ggiesen** [orbitalquark/textadept#493] [Ideas]
I'm a recent convert to Linux from Windows where I used Notepad++. One killer feature of it was the ability to autosave unsaved files (without prompting) and restore them on reopening. By unsaved files, I mean files that have never been saved anywhere. I frequently used Notepad++ as a "scratchpad" to jot down notes, config snippets, etc and would like to duplicate this functionality in Textadept.

Is such a thing possible with Textadept? If not, would it be possible to implement? Elementary Code has such a feature but the rest of the app is pretty basic, and other apps that implement it like Sublime Text are (IMHO) too heavyweight as a general text editor.

**Comment by orbitalquark** [comment]
This is not currently a feature, but it could be. I'll make a note to explore it as a possibility. Thanks for the suggestion.

**Comment by eric1234** [comment]
I wonder if this could be implemented in the `init.lua`? Have an event (timeout, change focus to another buffer, etc) that calls [`save_as`](https://orbitalquark.github.io/textadept/api.html#buffersave_asfilename) to a randomly generated temp file. Then override the save shortcut and menu to call your own function. That function would check if the file is currently saved to the temp location. If it is then call `save_as` again with no argument so it prompt the user for the path. If not (already a regular file) it delegates to the normal `save` method.

**Comment by orbitalquark** [comment]
I've created a new module for this feature: https://github.com/orbitalquark/textadept-scratch. I'm thinking to distribute it alongside Textadept releases in the separate modules.zip archive.

**Reply by ggiesen** [reply]
That's awesome! I'll try it out and report back

**Reply by Mirodin** [reply]
I did not know I want this even though I use Notepad++ and this exact feature regularly at work :+1:

**Reply by ggiesen** [reply]
This seems to work beautifully! Thank you very much.

Edit: I did find one bug. If you use both textadept-curses and textadept-gtk, and you close the scratch buffers on one, it will wipe out the scratch buffers for the other.

**Reply by snoopy** [reply]
@orbitalquark 
Would it be possible to add an index to untitled buffers so they can be more easily distinguished?

**Reply by orbitalquark** [reply]
It is possible, but there's the added complication of keeping a persistent counter across sessions, and when to reset such a counter. Also, it might also require some core modifications for displaying the name of untitled buffers. I'd rather keep things contained in this module for now.

Also, since tab reorder is possible, the user does have some degree of control on which scratch buffer is which.
